pkg/aws/client/client.go

```diff
@@ -765,6 +765,7 @@ func (c *Client) CreateVpc(ctx context.Context, desired *VPC) (*VPC, error) {
 		CidrBlock:                   aws.String(desired.CidrBlock),
 		AmazonProvidedIpv6CidrBlock: aws.Bool(desired.AssignGeneratedIPv6CidrBlock),
 		TagSpecifications:           desired.ToTagSpecifications(ec2types.ResourceTypeVpc),
+		InstanceTenancy:             ec2types.Tenancy(*desired.InstanceTenancy),
```

Comment:
As you already mentioned, the `InstanceTenancy *string` in the `VPC` struct in `pkg/aws/client/types.go` already existed and using it requires this conversion. I would change it to `InstanceTenancy *types.Tenancy` (from https://github.com/aws/aws-sdk-go-v2/tree/main/service/ec2/types). This ensures type safety and is easier to handle/read.

Reply:
Done. I wasn't sure about what happens when migrating existing terraform state to flow and if there was a reason why this property was introduced as ` *string`. Also not entirely sure how to test if this is an issue?